Update to net472 and net.sdk

milbrandt opened the pull request:
- Updated Project files to new Project Format style
- Updated from .NET Framework 4.5 to 4.7.2 (not going to dotnet Core as FSharp Compiler.Service.* is not available therefore)
- added NUnit TestAdapter for Visual Studio 2017 as Nuget package
- updated most Nuget packages (all except for FSharp Compiler.Service.* and FSharpLint.Core)

Open points:

- [x] ~several links are broken in sonar-fsharp-plugin\src\main\resources\org\sonar\l10n\fsharp\rules\fsharplint~ Issue #49
- [x] ~junit tests results are not reported in AppVeyor (only 21 F# Tests, 1 Junit test is missing)~ Issue #50
- [x] ~\[fs-\]analysis-(input|output).xml - which ones are needed, all required available?~ Issue #51
- [x] ~pom files parent references~ fixed in PR #48
- [x] ~Adjustments required to handle new rules ofFSharpLint (F# solution, Java wrapper plugin)~ Issue #64

Test:
- [x] @jmecosta 
- [x] @milbrandt - Sonarqube 7.7 developer Edition, Windows 7

Reminder: **DO SQUASH  MERGE**

milbrandt committed:
Enter commit messageupdate to net.Sdk project format

milbrandt committed:
Update NUnit framework

milbrandt committed:
update several nugets

milbrandt committed:
set FsSonarRunner as initial project

milbrandt committed:
update further nugets

milbrandt committed:
TargetFramework net472

milbrandt committed:
update version to 1.1 (new target framework)

milbrandt committed:
minor fix (path to solution directory)

milbrandt committed:
update remaining nuget packages

milbrandt committed:
update Namespace of FSharp.Compiler.Service 27.0.1

milbrandt committed:
some adjustments to updated FSharp.Compiler.Service/FSharpLint

jmecosta commented:
Think we need to pick another image in appveyour? Says cannot build some projects

jmecosta commented:
The yaml file is having os vs 2015. Supose we can bump that to vs 2019 direct

milbrandt commented:
> The yaml file is having os vs 2015. Supose we can bump that to vs 2019 direct

Would be fine for me. The current solution file is set for VS 2017, but I saw that the latest dotnet 2.2.300 requires VS 2019.

jmecosta commented:
update in master now

On Fri, May 31, 2019 at 10:20 AM Volker Milbrandt <notifications@github.com>
wrote:

> The yaml file is having os vs 2015. Supose we can bump that to vs 2019
> direct
>
> Would be fine for me. The current solution file is set for VS 2017, but I
> saw that the latest dotnet 2.2.300 requires VS 2019.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/jmecsoftware/sonar-fsharp-plugin/pull/30?email_source=notifications&email_token=AAH6HSB5LI6TADQ2CYHPGADPYDGUBA5CNFSM4HRBR3N2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODWUNONA#issuecomment-497604404>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAH6HSBEAFKXYCDCNC4SQCLPYDGUBANCNFSM4HRBR3NQ>
> .
>

milbrandt committed:
remove some dependencies which are transient included

jmecosta commented:
@milbrandt please rebase

milbrandt committed:
convert string to ISourceText

milbrandt committed:
update FSharpLint.Core because of FSharpLint.Application.XmlConfiguration.Configuration.defaultConfiguration

milbrandt committed:
namespace FSharpLint.Framework.Configuration replaced by FSharpLint.Application.XmlConfiguration.Configuration

milbrandt committed:
get Analyser configuration up to defer FSharpLint rework

FSharpLint PR https://github.com/fsprojects/FSharpLint/pull/330
See issue #35 here

milbrandt committed:
compiles, but at least one unit test still broken

Test Name:	RunLintInSource
Test FullName:	FsSonarRunnerCore.Test.TestLintRunner.RunLintInSource
Test Source:	D:\_Repos\sonar-fsharp-plugin_master\FsSonarRunner\FsSonarRunner.Test\TestLintRunner.fs : line 20
Test Outcome:	Failed
Test Duration:	0:00:00,062

Result StackTrace:
bei System.Resources.ManifestBasedResourceGroveler.HandleResourceStreamMissing(String fileName)
   bei System.Resources.ManifestBasedResourceGroveler.GrovelForResourceSet(CultureInfo culture, Dictionary`2 localResourceSets, Boolean tryParents, Boolean createIfNotExists, StackCrawlMark& stackMark)
   bei System.Resources.ResourceManager.InternalGetResourceSet(CultureInfo requestedCulture, Boolean createIfNotExists, Boolean tryParents, StackCrawlMark& stackMark)
   bei System.Resources.ResourceManager.InternalGetResourceSet(CultureInfo culture, Boolean createIfNotExists, Boolean tryParents)
   bei System.Resources.ResourceManager.GetResourceSet(CultureInfo culture, Boolean createIfNotExists, Boolean tryParents)
   bei FsSonarRunnerCore.SonarRules..ctor() in D:\_Repos\sonar-fsharp-plugin_master\FsSonarRunner\FsSonarRunnerCore\FsLintRunner.fs:Zeile 23.
   bei FsSonarRunnerCore.Test.TestLintRunner.RunLintInSource() in D:\_Repos\sonar-fsharp-plugin_master\FsSonarRunner\FsSonarRunner.Test\TestLintRunner.fs:Zeile 22.
Result Message:	System.Resources.MissingManifestResourceException : Für die angegebene Kultur oder die neutrale Kultur konnten keine Ressourcen gefunden werden. Stellen Sie sicher, dass Text.resources beim Kompilieren richtig in die Assembly FSharpLint.Core eingebettet wurde, oder dass die erforderlichen Satellitenassemblys geladen werden können und vollständig signiert sind.

milbrandt committed:
fix after rebase, update version info with alpha tag

milbrandt committed:
[wip] broken unit test

milbrandt committed:
fix and extend unit test

get FSharpLint resource manager

milbrandt commented on FsSonarRunner/FsSonarRunnerCore/FsLintRunner.fs:
```diff
@@ -19,7 +19,14 @@ type FsLintRule(name : string, value : string) =
 
     let fsLintProfile =
-        let resourceManager = new ResourceManager("Text" ,Assembly.Load("FSharpLint.Core"))
+        let resourceManager =
```
@jmecosta more complex, but I didn't got it runing otherwise. Maybe you know how to simplify it again?

milbrandt committed:
delete nuget from repo

milbrandt commented:
use dotnet build; dotnet test to build the VisualStudio solution

jmecosta commented:
i think you can make those changes directly to yaml file? i was about to suggest u to use a new version of nuget.exe to fix the nuger restore issues.

milbrandt commented:
> i think you can make those changes directly to yaml file? i was about to suggest u to use a new version of nuget.exe to fix the nuger restore issues.

No Need to use nuget.exe at all. dotnet build does it already. Yes, I think either in the yaml or at least in the pom file it will be possible. But it was already late yesterday night.

milbrandt committed:
fixes based on analysis with FsSonarRunner

milbrandt committed:
cosmetics

milbrandt committed:
dotnet build and test

milbrandt committed:
add Appveyor.TestLogger for vstest

milbrandt committed:
regenerate text files containing all rules and publishing it as artifact

milbrandt committed:
smaller adjustments

jmecosta commented:
@milbrandt after solving the conflicts this should be ready to merge? and thanks a lot for this massive update

milbrandt commented:
@jmecosta Unfortunately not yet :( There is still an issue with the plugin build in the pom. ANd then we have to test - it is update with lots of breaking changes in the dependencies.

But on the bright side: the number of rules have increased in FsharpLint and I could create some additional PRs on other projects

jmecosta commented:
sigh 
![image](https://user-images.githubusercontent.com/1041352/59151774-9e9b5c80-8a41-11e9-861a-dd27a9a3c7ce.png) how i would love if github would allow to reply in context

jmecosta commented:
> @jmecosta Unfortunately not yet :( There is still an issue with the plugin build in the pom. ANd then we have to test - it is update with lots of breaking changes in the dependencies.
> 
> But on the bright side: the number of rules have increased in FsharpLint and I could create some additional PRs on other projects

ok no problem, ive reference this https://github.com/TrimbleSolutionsCorporation/VSSonarQubeExtension/issues/278 to upgrade the linter there to last version and i will do all the testing then in our internal servers also!

milbrandt committed:
trigger maven build

milbrandt commented:
@jmecosta First test should be possible - but be aware, I also have it not deployed to any server yet.
I'm already aware of some smaller issues (which are also there in the master branch). I will list them in the initial comment of this PR for now.

milbrandt committed:
fix rebase issues

milbrandt committed:
merge issues & stash

milbrandt committed:
find location of JUnit results

milbrandt commented:
@jmecosta and everyone else: Any comments and help on open points are welcome. Especially on the two last ones, which are not so obvious.

jmecosta commented:
i would for now concetrant on net472 and net.sdk. we can have open issues for the items u suggest. this pr starts to be quite big for review.  if u can open issues for those  can discuss them there. But confused on what is the issue

jmecosta commented:
can we also handle the new rules in a dedicated pr please. I think its fine to review this pr and merge as is

jmecosta commented on FsSonarRunner/FsSonarRunner/App.config:
```diff
@@ -1,18 +0,0 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
```
we might needed these redirects. likely the option is disable now in project file. auto genererate bind redirects should be set in project.

milbrandt replied:
At least I was able to run `FsSonarRunner` on `FsSonarRunner.sln` and fixed the issues found in commit 7048ca2ba01a940bc171f391de1051b80b310f56 straightaway.

jmecosta commented on FsSonarRunner/FsSonarRunner/FsSonarRunner.fsproj:
```diff
@@ -1,124 +1,34 @@
-    </PropertyGroup>
-    <Error Condition="!Exists('..\packages\FSharp.Compiler.Service.ProjectCracker.13.0.0\build\net45\FSharp.Compiler.Service.ProjectCracker.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\FSharp.Compiler.Service.ProjectCracker.13.0.0\build\net45\FSharp.Compiler.Service.ProjectCracker.targets'))" />
+
+  <Target Name="PostBuild" AfterTargets="PostBuildEvent">
```
arent these already run regardless if the project needs to compile, i have always been against pre/post build events.

milbrandt replied:
According to [Microsoft](https://docs.microsoft.com/en-us/cpp/build/how-to-use-build-events-in-msbuild-projects?view=vs-2019)  "the _post-build_ event occurs after the build successfully ends."

For sure, this can also be done in appveyor. My goal was to publish the currently supported rule set with each build, as I saw thatupdating FSharpLint changed it.

milbrandt commented:
> i would for now concetrant on net472 and net.sdk. we can have open issues for the items u suggest. this pr starts to be quite big for review.  if u can open issues for those  can discuss them there. But confused on what is the issue

Ok, three issues created. The last one regarding changed rules of FSharpLint we Need to test or fix. Have you already installed an CI build of this branch on your server?

My next steps will be: tomorrow I'll put build of master in production, if that runs fine I'll give this branch a try.

milbrandt commented:
> can we also handle the new rules in a dedicated pr please. I think its fine to review this pr and merge as is

I'll fix the build straight away - and then we should test this. Can you please put it also on your Sonar Server so that we can test stability on more than one System?

jmecosta commented on sonar-fsharp-plugin/pom.xml:
```diff
@@ -6,7 +6,7 @@
   <parent>
```
could u had a section with developers and include yourself there

  <developers>
    <developer>
      <id></id>
      <name></name>
      <email></email>
    </developer>

milbrandt replied:
Thanks, the section is in the parent `pom.xml` This is inheritted, in the SonarQube marketplace the organization jmecsoftware is already displayed.

jmecosta reviewed (dismissed):
some minor things, rest LGTM

milbrandt committed:
fix build

jmecosta commented:
Yep, i try to push it during weekend.

On Thu, 13 Jun 2019, 23:27 Volker Milbrandt, <notifications@github.com>
wrote:

> can we also handle the new rules in a dedicated pr please. I think its
> fine to review this pr and merge as is
>
> I'll fix the build straight away - and then we should test this. Can you
> please put it also on your Sonar Server so that we can test stability on
> more than one System?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/jmecsoftware/sonar-fsharp-plugin/pull/30?email_source=notifications&email_token=AAH6HSCA2CJFR3BILH4WVCLP2KUULA5CNFSM4HRBR3N2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODXU565Y#issuecomment-501866359>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAH6HSBPCW3ZYLQBU6BAFETP2KUULANCNFSM4HRBR3NQ>
> .
>

jmecosta commented:
Think it works because u have fsharp.core 4.3.0 0 in cache. Fsharp data has
dependecny on that so it will probably fail if that is not found in gac

On Fri, 14 Jun 2019, 00:01 Volker Milbrandt, <notifications@github.com>
wrote:

> *@milbrandt* commented on this pull request.
> ------------------------------
>
> In FsSonarRunner/FsSonarRunner/App.config
> <https://github.com/jmecsoftware/sonar-fsharp-plugin/pull/30#discussion_r293575062>
> :
>
> > @@ -1,18 +0,0 @@
>
> -﻿<?xml version="1.0" encoding="utf-8"?>
>
>
> At least I was able to run FsSonarRunner on FsSonarRunner.sln and fixed
> the issues found in commit 7048ca2
> <https://github.com/jmecsoftware/sonar-fsharp-plugin/commit/7048ca2ba01a940bc171f391de1051b80b310f56>
> straightaway.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/jmecsoftware/sonar-fsharp-plugin/pull/30?email_source=notifications&email_token=AAH6HSEA34J7IDWQ2KT3TSTP2KYT5A5CNFSM4HRBR3N2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOB3QKHXQ#discussion_r293575062>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAH6HSDSXO5BSE5BWZF2NXDP2KYT5ANCNFSM4HRBR3NQ>
> .
>

milbrandt committed:
fix rebase

milbrandt committed:
dotnet core project files patching

milbrandt committed:
redundant new

milbrandt committed:
exception thrown when option /d with one argument

milbrandt committed:
revert (unit test broken)

milbrandt commented:
**1.1.0.159-rc** Run code analysis fails with:
>2019-06-21T19:54:28.4314285Z 21:54:28.416 INFO: Sensor F# [sonarqubefsharpplugin]
2019-06-21T19:54:28.5093646Z 21:54:28.494 DEBUG: Unzip D:\_2\8\.sonarqube\out\.sonar\mod1\ProjectTools\cs
2019-06-21T19:54:28.5094996Z 21:54:28.494 DEBUG: Unzip D:\_2\8\.sonarqube\out\.sonar\mod1\ProjectTools\de
2019-06-21T19:54:28.5097111Z 21:54:28.494 DEBUG: Unzip D:\_2\8\.sonarqube\out\.sonar\mod1\ProjectTools\en
2019-06-21T19:54:28.5098950Z 21:54:28.494 DEBUG: Unzip D:\_2\8\.sonarqube\out\.sonar\mod1\ProjectTools\es
2019-06-21T19:54:28.5101327Z 21:54:28.494 DEBUG: Unzip D:\_2\8\.sonarqube\out\.sonar\mod1\ProjectTools\fr
2019-06-21T19:54:28.5103265Z 21:54:28.494 DEBUG: Unzip D:\_2\8\.sonarqube\out\.sonar\mod1\ProjectTools\it
2019-06-21T19:54:28.5105519Z 21:54:28.494 DEBUG: Unzip D:\_2\8\.sonarqube\out\.sonar\mod1\ProjectTools\ja
2019-06-21T19:54:28.5107463Z 21:54:28.510 DEBUG: Unzip D:\_2\8\.sonarqube\out\.sonar\mod1\ProjectTools\ko
2019-06-21T19:54:28.5148633Z 21:54:28.510 DEBUG: Unzip D:\_2\8\.sonarqube\out\.sonar\mod1\ProjectTools\pl
2019-06-21T19:54:28.5150473Z 21:54:28.510 DEBUG: Unzip D:\_2\8\.sonarqube\out\.sonar\mod1\ProjectTools\pt-BR
2019-06-21T19:54:28.5152124Z 21:54:28.510 DEBUG: Unzip D:\_2\8\.sonarqube\out\.sonar\mod1\ProjectTools\ru
2019-06-21T19:54:28.5153962Z 21:54:28.510 DEBUG: Unzip D:\_2\8\.sonarqube\out\.sonar\mod1\ProjectTools\tr
2019-06-21T19:54:28.5155487Z 21:54:28.510 DEBUG: Unzip D:\_2\8\.sonarqube\out\.sonar\mod1\ProjectTools\zh-Hans
2019-06-21T19:54:28.5157655Z 21:54:28.510 DEBUG: Unzip D:\_2\8\.sonarqube\out\.sonar\mod1\ProjectTools\zh-Hant
2019-06-21T19:54:28.5159342Z 21:54:28.510 DEBUG: Unzip D:\_2\8\.sonarqube\out\.sonar\mod1\ProjectTools\cs\FSharp.Core.resources.dll
2019-06-21T19:54:28.5173567Z ##[error]21:54:28.510 ERROR: Unzip Failed D:\_2\8\.sonarqube\out\.sonar\mod1\ProjectTools\cs\FSharp.Core.resources.dll (The system cannot find the path specified)
2019-06-21T19:54:28.5174040Z 21:54:28.510 ERROR: Unzip Failed D:\_2\8\.sonarqube\out\.sonar\mod1\ProjectTools\cs\FSharp.Core.resources.dll (The system cannot find the path specified)
2019-06-21T19:54:28.5174961Z ##[error]21:54:28.510 ERROR: Unable to unzip File: FsSonarRunner.exe => D:\_2\8\.sonarqube\out\.sonar\mod1\ProjectTools\cs\FSharp.Core.resources.dll (The system cannot find the path specified)
21:54:28.510 ERROR: Could not write settings to file '{0}'
2019-06-21T19:54:28.5175270Z 21:54:28.510 ERROR: Unable to unzip File: FsSonarRunner.exe => D:\_2\8\.sonarqube\out\.sonar\mod1\ProjectTools\cs\FSharp.Core.resources.dll (The system cannot find the path specified)
2019-06-21T19:54:28.5175412Z 21:54:28.510 ERROR: Could not write settings to file '{0}'
2019-06-21T19:54:28.5182057Z ##[error]21:54:28.510 ERROR: Not able to parse file : {0}
2019-06-21T19:54:28.5182357Z 21:54:28.510 ERROR: Not able to parse file : {0}
2019-06-21T19:54:28.5182913Z 21:54:28.510 INFO: Sensor F# [sonarqubefsharpplugin] (done) | time=94ms
2019-06-21T19:54:28.5201089Z 21:54:28.510 INFO: ------------- Run sensors on module WindowsService

milbrandt commented:
**1.1.0.159-rc** Displayname in server Marketplace changed and not ok:
![Anmerkung 2019-06-21 220244](https://user-images.githubusercontent.com/26173245/59948440-5d0ea680-9470-11e9-914a-5aeaf0687b90.png)

milbrandt committed:
change name in "Update Center"

milbrandt commented:
> 1.1.0.159-rc Displayname in server Marketplace changed and not ok:

fixed in 1.1.0.163-rc

milbrandt committed:
Unzip an archive with sub-directories and extract its contents

milbrandt commented:
> 1.1.0.159-rc Run code analysis fails with:
> 
> 2019-06-21T19:54:28.4314285Z 21:54:28.416 INFO: Sensor F# [sonarqubefsharpplugin]
> 2019-06-21T19:54:28.5093646Z 21:54:28.494 DEBUG: Unzip D:_2\8.sonarqube\out.sonar\mod1\ProjectTools\cs
> 2019-06-21T19:54:28.5094996Z 21:54:28.494 DEBUG: Unzip D:_2\8.sonarqube\out.sonar\mod1\ProjectTools\de
> 2019-06-21T19:54:28.5097111Z 21:54:28.494 DEBUG: Unzip D:_2\8.sonarqube\out.sonar\mod1\ProjectTools\en
> 2019-06-21T19:54:28.5098950Z 21:54:28.494 DEBUG: Unzip D:_2\8.sonarqube\out.sonar\mod1\ProjectTools\es
> 2019-06-21T19:54:28.5101327Z 21:54:28.494 DEBUG: Unzip D:_2\8.sonarqube\out.sonar\mod1\ProjectTools\fr
> 2019-06-21T19:54:28.5103265Z 21:54:28.494 DEBUG: Unzip D:_2\8.sonarqube\out.sonar\mod1\ProjectTools\it
> 2019-06-21T19:54:28.5105519Z 21:54:28.494 DEBUG: Unzip D:_2\8.sonarqube\out.sonar\mod1\ProjectTools\ja
> 2019-06-21T19:54:28.5107463Z 21:54:28.510 DEBUG: Unzip D:_2\8.sonarqube\out.sonar\mod1\ProjectTools\ko
> 2019-06-21T19:54:28.5148633Z 21:54:28.510 DEBUG: Unzip D:_2\8.sonarqube\out.sonar\mod1\ProjectTools\pl
> 2019-06-21T19:54:28.5150473Z 21:54:28.510 DEBUG: Unzip D:_2\8.sonarqube\out.sonar\mod1\ProjectTools\pt-BR
> 2019-06-21T19:54:28.5152124Z 21:54:28.510 DEBUG: Unzip D:_2\8.sonarqube\out.sonar\mod1\ProjectTools\ru
> 2019-06-21T19:54:28.5153962Z 21:54:28.510 DEBUG: Unzip D:_2\8.sonarqube\out.sonar\mod1\ProjectTools\tr
> 2019-06-21T19:54:28.5155487Z 21:54:28.510 DEBUG: Unzip D:_2\8.sonarqube\out.sonar\mod1\ProjectTools\zh-Hans
> 2019-06-21T19:54:28.5157655Z 21:54:28.510 DEBUG: Unzip D:_2\8.sonarqube\out.sonar\mod1\ProjectTools\zh-Hant
> 2019-06-21T19:54:28.5159342Z 21:54:28.510 DEBUG: Unzip D:_2\8.sonarqube\out.sonar\mod1\ProjectTools\cs\FSharp.Core.resources.dll
> 2019-06-21T19:54:28.5173567Z ##[error]21:54:28.510 ERROR: Unzip Failed D:_2\8.sonarqube\out.sonar\mod1\ProjectTools\cs\FSharp.Core.resources.dll (The system cannot find the path specified)
> 2019-06-21T19:54:28.5174040Z 21:54:28.510 ERROR: Unzip Failed D:_2\8.sonarqube\out.sonar\mod1\ProjectTools\cs\FSharp.Core.resources.dll (The system cannot find the path specified)
> 2019-06-21T19:54:28.5174961Z ##[error]21:54:28.510 ERROR: Unable to unzip File: FsSonarRunner.exe => D:_2\8.sonarqube\out.sonar\mod1\ProjectTools\cs\FSharp.Core.resources.dll (The system cannot find the path specified)
> 21:54:28.510 ERROR: Could not write settings to file '{0}'
> 2019-06-21T19:54:28.5175270Z 21:54:28.510 ERROR: Unable to unzip File: FsSonarRunner.exe => D:_2\8.sonarqube\out.sonar\mod1\ProjectTools\cs\FSharp.Core.resources.dll (The system cannot find the path specified)
> 2019-06-21T19:54:28.5175412Z 21:54:28.510 ERROR: Could not write settings to file '{0}'
> 2019-06-21T19:54:28.5182057Z ##[error]21:54:28.510 ERROR: Not able to parse file : {0}
> 2019-06-21T19:54:28.5182357Z 21:54:28.510 ERROR: Not able to parse file : {0}
> 2019-06-21T19:54:28.5182913Z 21:54:28.510 INFO: Sensor F# [sonarqubefsharpplugin] (done) | time=94ms
> 2019-06-21T19:54:28.5201089Z 21:54:28.510 INFO: ------------- Run sensors on module WindowsService

fixed in 1.1.0.166-rc

milbrandt commented:
1.1.0-166-rc: reports on 2 files in a solution:

> 2019-06-22T19:34:49.9139899Z 21:34:49.898 INFO: Not Supported member Report
2019-06-22T19:34:50.5440443Z 21:34:50.539 INFO: Lint Ok

milbrandt commented:
1.1.0.166-rc issues not reported to Sonar:

Manual execution:
>D:\_2\8\.sonarqube\out\.sonar\mod1\ProjectTools>FsSonarRunner.exe /f D:/_2/8/s/s
ource/WindowsService/Program.fs
[|"/f"; "D:/_2/8/s/source/WindowsService/Program.fs"|]
Lint Ok
D:/_2/8/s/source/WindowsService/Program.fs : RulesRedundantNewKeyword : 31 : FL0
014: Usage of `new` keyword here is redundant.
D:/_2/8/s/source/WindowsService/Program.fs : RulesRedundantNewKeyword : 25 : FL0
014: Usage of `new` keyword here is redundant.

Execution on build:
>2019-06-22T19:34:44.5654885Z 21:34:44.555 DEBUG: D:\_2\8\.sonarqube\out\.sonar\mod1\ProjectTools\FsSonarRunner.exe /i:D:\_2\8\.sonarqube\out\.sonar\mod2\fs-analysis-input.xml /o:D:\_2\8\.sonarqube\out\.sonar\mod2\fs-analysis-output.xml
2019-06-22T19:34:44.7319228Z 21:34:44.727 INFO: [|"/i:D:\_2\8\.sonarqube\out\.sonar\mod2\fs-analysis-input.xml";
2019-06-22T19:34:44.7319456Z 21:34:44.727 INFO:   "/o:D:\_2\8\.sonarqube\out\.sonar\mod2\fs-analysis-output.xml"|]
...Some lines skipped...
2019-06-22T19:34:50.8823349Z 21:34:50.867 INFO: Write output xml to file D:\_2\8\.sonarqube\out\.sonar\mod2\fs-analysis-output.xml
2019-06-22T19:34:50.9051114Z 21:34:50.898 DEBUG: 'source/WindowsService/Program.fs' generated metadata with charset 'UTF-8'
2019-06-22T19:34:50.9095590Z 21:34:50.898 DEBUG: 'source/WindowsService/ServiceInstaller.fs' generated metadata with charset 'windows-1252'
2019-06-22T19:34:50.9123805Z 21:34:50.898 DEBUG: 'source/WindowsService/WindowsService.fs' generated metadata with charset 'windows-1252'
2019-06-22T19:34:50.9155469Z 21:34:50.914 INFO: Sensor F# [sonarqubefsharpplugin] (done) | time=6359ms

In `fs-analysis-input`:
```xml
    <Rule>
      <Key>RulesRedundantNewKeyword</Key>
      <Parameters>
      </Parameters>
    </Rule>
```

in `fs-analysis-output.xml`:
```xml
    <File>
      <Path>D:/_2/8/s/source/WindowsService/Program.fs</Path>
      <Metrics>
        <Classes>0</Classes>
        <Accessors>0</Accessors>
        <Functions>20</Functions>
        <Complexity>3</Complexity>
        <FileComplexityDistribution>0=1;5=0;10=0;20=0;30=0;60=0;90=0</FileComplexityDistribution>
        <FunctionComplexityDistribution>1=0;2=1;4=0;6=0;8=0;10=0;12=3</FunctionComplexityDistribution>
        <LinesOfCode>
...Several lines skipped...
          <Line>25</Line>
          <Line>26</Line>
          <Line>27</Line>
          <Line>29</Line>
          <Line>30</Line>
          <Line>31</Line>
...Several lines skipped...
        </LinesOfCode>
      </Metrics>
      <Issues />
      <CopyPasteTokens>
...Several lines skipped...
        <Token>
          <Value>Q29sbGVjdGlvbnM=</Value>
          <Line>25</Line>
          <LeftColoumn>25</LeftColoumn>
          <RightColoumn>36</RightColoumn>
        </Token>
        <Token>
          <Value>SGFzaHRhYmxl</Value>
          <Line>25</Line>
          <LeftColoumn>37</LeftColoumn>
          <RightColoumn>46</RightColoumn>
        </Token>
...Several lines skipped...
        <Token>
          <Value>bGV0</Value>
          <Line>31</Line>
          <LeftColoumn>4</LeftColoumn>
          <RightColoumn>7</RightColoumn>
        </Token>
        <Token>
          <Value>ZGlj</Value>
          <Line>31</Line>
          <LeftColoumn>8</LeftColoumn>
          <RightColoumn>11</RightColoumn>
        </Token>
        <Token>
          <Value>bmV3</Value>
          <Line>31</Line>
          <LeftColoumn>14</LeftColoumn>
          <RightColoumn>17</RightColoumn>
        </Token>
        <Token>
          <Value>U3lzdGVt</Value>
          <Line>31</Line>
          <LeftColoumn>18</LeftColoumn>
          <RightColoumn>24</RightColoumn>
        </Token>
        <Token>
          <Value>Q29sbGVjdGlvbnM=</Value>
          <Line>31</Line>
          <LeftColoumn>25</LeftColoumn>
          <RightColoumn>36</RightColoumn>
        </Token>
        <Token>
          <Value>SGFzaHRhYmxl</Value>
          <Line>31</Line>
          <LeftColoumn>37</LeftColoumn>
          <RightColoumn>46</RightColoumn>
        </Token>
...Several lines skipped...
      </CopyPasteTokens>
    </File>
```

milbrandt commented:
1.1.0.166-rc on build Server log:
>2019-06-22T19:35:10.2742547Z 21:35:10.263 DEBUG: Detection of duplications for D:/_2/8/s/source/Service/Program.fs

but duplicated lines in Sonar server shows
>source/Service/Program.fs | 0.0% | 0

milbrandt commented:
@jmecosta The current build 1.1.0.166 is running for me except for reporting issues with VS2017 on build server. Could you please also test and report with Sonarqube version and Visual Studio version on build server?

milbrandt committed:
change lib to netstandard2.0 and unit tests to netcoerapp2.2

milbrandt commented:
1.1.0.166-rc reported
> FL0034: If `GetApps` has no mutable arguments partially applied then the lambda can be removed
and in another project duplication on `fs` files are reported.

milbrandt committed:
remove direct dependency of FSharp.Compiler.Service (already transitive by FSharpLint.Core)

jmecosta commented:
> @jmecosta The current build 1.1.0.166 is running for me except for reporting issues with VS2017 on build server. Could you please also test and report with Sonarqube version and Visual Studio version on build server?

yep, i will! thanks.

milbrandt committed:
remove rules no longer supported from FSharpLint

jmecosta commented:
build locally: 
[INFO] Reading assembly descriptor: zip-assembly.xml
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for SonarQube F# Plugin 1.1.0-rc:
[INFO]
[INFO] SonarQube F# Plugin ................................ SUCCESS [ 12.983 s]
[INFO] Sonar F# Plugin :: FsSonarRunner ................... FAILURE [  4.599 s]
[INFO] Sonar F# Plugin :: Core ............................ SKIPPED
[INFO] ------------------------------------------------------------------------

Looks like there is no more run in pom.xml to build the solution

milbrandt committed:
fix build

milbrandt commented:
> Looks like there is no more run in pom.xml to build the solution

No, Java broken - removed initialization but used 2 variables.

milbrandt commented on sonar-fsharp-plugin/src/main/resources/com/sonar/sqale/fsharp-fslint-model.xml:
```diff
@@ -117,19 +117,6 @@
                     <txt>mn</txt>
                 </prop>
             </chc>
-            <chc>
```
@jmecosta Do you remember for which reason this file is required? Didn't find where it is used.

jmecosta replied:
that's the old sqale model file, it provides basic values for calculation of technical debt. most likely those or part of those are decrecated with the introduction of the smell/bug/vulnerability but I have checked how other plugins are doing that. For now it seems it will still take the estimates and if it works I suggest leave it like that for now. needs proper checking later

milbrandt replied:
I was curious as the removed rules where in this file - and there are lots of rules which we will need to introduce tofully support the current FSharpLint.

milbrandt committed:
rules are supported by FSharpLint, not deprecated

milbrandt committed:
rules not supported by FSharpLint

jmecosta commented:
> Looks like there is no more run in pom.xml to build the solution
> 
> No, Java broken - removed initialization but used 2 variables.

? so how do I build the all thing locally without manually building the core first

milbrandt commented:
>>> Looks like there is no more run in pom.xml to build the solution

>> No, Java broken - removed initialization but used 2 variables.

> ? so how do I build the all thing locally without manually building the core first

Sorry, that was a misunderstanding - I thought you meant the broken build.
It's correct, dotnet build is executed by `appveyor.yml` and not from the pom. pom only zips the dotnet build atrifact and build the Java plugin.

milbrandt committed:
rework of argument evaluation

- new option /version
- option /o supported for /i, /f and /d

milbrandt commented:
> 1.1.0-166-rc: reports on 2 files in a solution:
> 
> 2019-06-22T19:34:49.9139899Z 21:34:49.898 INFO: Not Supported member Report
> 2019-06-22T19:34:50.5440443Z 21:34:50.539 INFO: Lint Ok

uncritical, can be handled in separate PR

milbrandt commented:
> 1.1.0.166-rc on build Server log:
> 
> 2019-06-22T19:35:10.2742547Z 21:35:10.263 DEBUG: Detection of duplications for D:/_2/8/s/source/Service/Program.fs
> 
> but duplicated lines in Sonar server shows
> 
> source/Service/Program.fs | 0.0% | 0

split in separate issue

milbrandt committed:
fix of "Not Supported member Report" #68 can be delayed, not critical

milbrandt committed:
synchronize rules between Java and F#

milbrandt committed:
build dotnet from maven

milbrandt committed:
trigger build

milbrandt committed:
fix build

milbrandt committed:
another try to fix build

milbrandt committed:
try to use configuration again

milbrandt committed:
failed to start SonarQube

milbrandt committed:
dotnet test results

milbrandt committed:
failed to start SonarQube

milbrandt commented:
>>>> Looks like there is no more run in pom.xml to build the solution

>> ? so how do I build the all thing locally without manually building the core first

Now with the updated pom files you can build locally everything with Maven again, with configuration Debug or Release:

    mvn -Dconfiguration=Debug clean package

milbrandt commented:
in 190-rc 2 issues are reported in `.sonarqube\out\.sonar\mod2\fs-analysis-output.xml` but not reported in SonarQube server

Update: Issue is handled in `FSharpSensor.AnalysisResultImporter.handleIssueTag` (log message is printed).
The error key is not included in `sonar-fsharp-plugin\src\main\resources\com\sonar\sqale\fsharp-fslint-model.xml`

Update 2: fixed in 192-rc

milbrandt committed:
fix report of RulesRedundantNewKeyword

milbrandt committed:
typo

milbrandt committed:
additional trace logging

milbrandt commented:
@jmecosta Build 193-rc I installed on our server in production. If you could verify on your side that no crash occurs and some reports are published on Sonarqube Server I would like to squash merge this PR. On the next days I'll observe our server regarding F# builds.

jmecosta commented:
compilation now works fine thanks, im installed the plugin in server and run analysis all working perfectly fine. Thanks

jmecosta commented:
good to be merged

milbrandt committed:
update nuget

milbrandt committed:
Merge branch 'master' into netSdkProjectFormat

# Conflicts:
#	FsSonarRunner/FsSonarRunner/AssemblyInfo.fs
#	FsSonarRunner/FsSonarRunnerCore.Test/AssemblyInfo.fs
#	FsSonarRunner/FsSonarRunnerCore/AssemblyInfo.fs
#	FsSonarRunner/pom.xml
#	appveyor.yml
#	pom.xml
#	sonar-fsharp-plugin/pom.xml

milbrandt committed:
error handling on test method, remove rc flag
